implementations/python/examples/08.py

```diff
@@ -12,23 +12,29 @@ async def main(node):
         node=node,
         name="Assistant",
         instructions="Assistant to solve some complex task ...",
-        model=Model(name="ollama_chat/llama3.2"),
-        planner=CoTPlanner(),
+        model=Model(name="deepseek-r1"),
+        planner=CoTPlanner(model=Model(name="deepseek-r1")),
     )
 
     async for chunk in agent.send_stream("Estimate how many violins are in the world", timeout=60 * 20):
         if len(chunk.snippet.messages) == 0:
             continue
 
         phase = chunk.snippet.messages[-1].phase
+        thinking = chunk.snippet.messages[-1].thinking
         text = chunk.snippet.messages[-1].content
 
-        if phase == Phase.THINKING:
-            print(f"\033[94m{text}\033[0m", end="")  # thinking: blue
-        elif phase == Phase.PLANNING:
-            print(f"\033[92m{text}\033[0m", end="")  # planning: green
-        elif phase == Phase.EXECUTING:
-            print(f"\033[90m{text}\033[0m", end="")  # executing: grey
+        # reasoning is always in italic
+        if phase == Phase.PLANNING:  # planning: green
+            if thinking:
+                print(f"\033[3;92m{text}\033[0m", end="")
```

Comment:
We should use `info` instead of `print` because this allows us to control the output centrally if necessary.

Reply:
This is printing a piece of text in streaming mode, using logging would completely break the output